src/coerce_string_value.ts

```diff
@@ -0,0 +1,31 @@
+const hasDotRegexp = /\./gi;
+
+// tslint:disable-next-line: cyclomatic-complexity
+export function coerceStringValue(value: string) {
+    /**
+     * Only try casting to float if there's at least one `.`
+     *
+     * This MUST come before parseInt because parseInt will succeed to
+     * parse a float but it will be lossy, e.g.
+     * parseInt('1.24242', 10) === 1
+     */
```

Comment:
Will fail below:
` parseFloat('1.0 version') => 1 `

Possibly run isNaN(value) first then check if 'null','false','true', then parseInt/parseFloat.

Reply:
doesn't the second part of the && expression protect from this?

Reply:
Oh wait so you're saying it will succeed to parse that string because it _starts_ with a float?

Reply:
yes, same with parseInt.
`parseInt('3Com',10) = 3`

Reply:
This is horrible.

Reply:
Updated to use `Number` constructor